### PR TITLE
[somfytahoma] Setting of channels at init + UoM for channels

### LINKED ISSUE #10285: [somfytahoma] Use QuantityType to set channels linked to item of type Number with a dimension
As I already mentioned in another place, the binding defines channels for items of type Number with a dimension like for example `Number:Temperature` but these channels are set using a simple `DecimalType` without specifying any unit. A `QuantityType` should be used and the unit should be set.

Here are the list of channel types concerned:

1. luminance => Number:Illuminance - used only by the thing type lightsensor
2. current_temperature => Number:Temperature - used only by the thing type heatingsystem
3. humidity => Number:Dimensionless - used only by the thing type humiditysensor
4. target_temperature => Number:Temperature - used by the thing types heatingsystem, thermostat and valveheatingsystem
5. temperature => Number:Temperature - used by the thing types temperaturesensor, thermostat and valveheatingsystem
6. energy_consumption => Number:Energy - used only by the thing type electricitysensor

@octa22 are the units adjustable somewhere for the Tahoma system ? Or are there fixed and the API can only return one unit for a particular state ?
Could you provide the unit used by the API for each API state used to set these 6 channels ?
We could then use a QuantityType to set the channel value rather than a DecimalType.

Could you identity too for which one the API is providing a device attribute `core:MeasuredValueType` defining the unit of the state ? I agree with you that it is apparently not always present but I see it for example for the temperature sensor (5th channel in my list) and the energy consumption sensor (6th channel in my list). This could be used at least when present.

### LINKED ISSUE #10291: [somfytahoma] Channels not set at startup
When the binding is started, the channels are not initialized. They are as soon as there is a change of the corresponding state.
Unfortunately, the binding is relying on the hypothesis that a REFRESH command will be sent at startup to all linked channels.
This is something that was done in OH2 but no more in OH3.
@octa22 : it is necessary to trigger the refresh of channels at startup. I am not yet sure what is the best way to do it considering the binding architecture. But very probably in the `initialize` method of class `SomfyTahomaBaseThingHandler`.

### PR DESCRIPTION
**Pull request by lolodomo**
Fix #10291
Fix #10285

Signed-off-by: Laurent Garnier <lg.hc@free.fr>

### CODE REVIEW

File: bundles/org.openhab.binding.somfytahoma/src/main/java/org/openhab/binding/somfytahoma/internal/handler/SomfyTahomaBaseThingHandler.java
Comment (octa22) [inline bundles/org.openhab.binding.somfytahoma/src/main/java/org/openhab/binding/somfytahoma/internal/handler/SomfyTahomaBaseThingHandler.java:240]: Missing case for case "core:TemperatureInCelsius" (with s in Celsius)

**Review by octa22 — CHANGES_REQUESTED** [review]
I have found a missing case in a switch, otherwise LGTM

File: bundles/org.openhab.binding.somfytahoma/src/main/java/org/openhab/binding/somfytahoma/internal/handler/SomfyTahomaBaseThingHandler.java
Comment (lolodomo) [reply bundles/org.openhab.binding.somfytahoma/src/main/java/org/openhab/binding/somfytahoma/internal/handler/SomfyTahomaBaseThingHandler.java:240]: Added.
I can confirm that with the Cozytouch I find a value with a c.

**Comment by lolodomo** [comment]
@octa22 : can you please approve now?

**Review by octa22 — APPROVED** [review]

**Comment by lolodomo** [comment]
@openhab/add-ons-maintainers : is it possible to have a review & merge of this PR which is now validated by the binding author ?

File: bundles/org.openhab.binding.somfytahoma/src/main/java/org/openhab/binding/somfytahoma/internal/handler/SomfyTahomaBaseThingHandler.java
Comment (fwolter) [inline bundles/org.openhab.binding.somfytahoma/src/main/java/org/openhab/binding/somfytahoma/internal/handler/SomfyTahomaBaseThingHandler.java:87]: Can this message be replaced by using the debugger or by increasing the framework's log level? See point 4 https://www.openhab.org/docs/developer/guidelines.html#f-logging

This is already logged [here](https://github.com/openhab/openhab-core/blob/e20348e8d66845db841f17e6e048d789d004fdf2/bundles/org.openhab.core.thing/src/main/java/org/openhab/core/thing/internal/ThingManagerImpl.java#L815) and can therefore be removed.

File: bundles/org.openhab.binding.somfytahoma/src/main/java/org/openhab/binding/somfytahoma/internal/handler/SomfyTahomaBaseThingHandler.java
Comment (fwolter) [inline bundles/org.openhab.binding.somfytahoma/src/main/java/org/openhab/binding/somfytahoma/internal/handler/SomfyTahomaBaseThingHandler.java:94]: The log message could be removed, as the status update is already logged by the framework.

The state change originated by updateStatus() is logged to events.log. Including the status detail message.

File: bundles/org.openhab.binding.somfytahoma/src/main/java/org/openhab/binding/somfytahoma/internal/handler/SomfyTahomaBaseThingHandler.java
Comment (fwolter) [inline bundles/org.openhab.binding.somfytahoma/src/main/java/org/openhab/binding/somfytahoma/internal/handler/SomfyTahomaBaseThingHandler.java:119]: Actually, these states are set by the framework, if the bridge is offline. Is there a reason why you set these explicitly?

**Review by fwolter — CHANGES_REQUESTED** [review]

File: bundles/org.openhab.binding.somfytahoma/src/main/java/org/openhab/binding/somfytahoma/internal/handler/SomfyTahomaBaseThingHandler.java
Comment (lolodomo) [reply bundles/org.openhab.binding.somfytahoma/src/main/java/org/openhab/binding/somfytahoma/internal/handler/SomfyTahomaBaseThingHandler.java:87]: I changed into `logger.trace`.

File: bundles/org.openhab.binding.somfytahoma/src/main/java/org/openhab/binding/somfytahoma/internal/handler/SomfyTahomaBaseThingHandler.java
Comment (lolodomo) [reply bundles/org.openhab.binding.somfytahoma/src/main/java/org/openhab/binding/somfytahoma/internal/handler/SomfyTahomaBaseThingHandler.java:94]: I changed into `logger.trace`.
This is a log just to see when `bridgeStatusChanged` is called.

File: bundles/org.openhab.binding.somfytahoma/src/main/java/org/openhab/binding/somfytahoma/internal/handler/SomfyTahomaBaseThingHandler.java
Comment (lolodomo) [reply bundles/org.openhab.binding.somfytahoma/src/main/java/org/openhab/binding/somfytahoma/internal/handler/SomfyTahomaBaseThingHandler.java:119]: The default behaviour of `bridgeStatusChanged` is to set the thing ONLINE when the bridge becomes ONLINE. That's all. In our case, we need to run additional tasks when it occurs like setting channels. That is the reason why `bridgeStatusChanged` is ovverriden. Because it is overriden, I have to handle all cases, including the case the bridge becomes OFFLINE.

File: bundles/org.openhab.binding.somfytahoma/src/main/java/org/openhab/binding/somfytahoma/internal/handler/SomfyTahomaBaseThingHandler.java
Comment (fwolter) [reply bundles/org.openhab.binding.somfytahoma/src/main/java/org/openhab/binding/somfytahoma/internal/handler/SomfyTahomaBaseThingHandler.java]: The log message is redundant and should therefore be removed. You could set the log level of `org.openhab.core.thing` to `DEBUG` to see the log message from the framework: "Calling initialize handler for thing '{}' at '{}'."

File: bundles/org.openhab.binding.somfytahoma/src/main/java/org/openhab/binding/somfytahoma/internal/handler/SomfyTahomaBaseThingHandler.java
Comment (fwolter) [reply bundles/org.openhab.binding.somfytahoma/src/main/java/org/openhab/binding/somfytahoma/internal/handler/SomfyTahomaBaseThingHandler.java]: You could use the debugger to see when the method is called. Here is an excerpt from the coding guidelines:

```
Logging is not a replacement for using the debugger.

void myFun() {
    logger.trace("Enter myfun"); // DONT, DONT, really DONT do that
    doSomething();
    logger.trace("Leave myfun"); // DONT, DONT, really DONT do that
}
```

File: bundles/org.openhab.binding.somfytahoma/src/main/java/org/openhab/binding/somfytahoma/internal/handler/SomfyTahomaBaseThingHandler.java
Comment (fwolter) [reply bundles/org.openhab.binding.somfytahoma/src/main/java/org/openhab/binding/somfytahoma/internal/handler/SomfyTahomaBaseThingHandler.java:119]: OK

File: bundles/org.openhab.binding.somfytahoma/src/main/java/org/openhab/binding/somfytahoma/internal/handler/SomfyTahomaBaseThingHandler.java
Comment (lolodomo) [reply bundles/org.openhab.binding.somfytahoma/src/main/java/org/openhab/binding/somfytahoma/internal/handler/SomfyTahomaBaseThingHandler.java]: But these bindiongs DEBUG logs like many others are logs that should help us, developers, to understand what's wrong with the binding when a user has a problem. We ask the user to enable the binding DEBUG logs and we can have more information.
If by debugger, you mean Eclipse debugger, of course I can use it myself but a standard user will never do it.

I will remove these logs but I do not understand the new philosophy with logs.

Comment (fwolter) [reply bundles/org.openhab.binding.somfytahoma/src/main/java/org/openhab/binding/somfytahoma/internal/handler/SomfyTahomaBaseThingHandler.java]: Redundant logging doesn't seem right, just to avoid setting an additional log level. If the state of the Thing is important for tracking down a problem, the logging for the Thing state should be enabled.

Comment (lolodomo) [reply bundles/org.openhab.binding.somfytahoma/src/main/java/org/openhab/binding/somfytahoma/internal/handler/SomfyTahomaBaseThingHandler.java]: This is largely easier when you don't have to know and enable multiple logs, including logs from the core framework. Even a binding developer is not necessarily aware of how is implemented the core framework and what packages the logs have to be enabled. Even for the binding developer, it will require to search in the code of the core framework !
IMHO, reducing logs at the binding level will just add more difficulty to analyze problems reported by users.

File: bundles/org.openhab.binding.somfytahoma/src/main/java/org/openhab/binding/somfytahoma/internal/handler/SomfyTahomaBaseThingHandler.java
Comment (lolodomo) [reply bundles/org.openhab.binding.somfytahoma/src/main/java/org/openhab/binding/somfytahoma/internal/handler/SomfyTahomaBaseThingHandler.java]: Log removed

File: bundles/org.openhab.binding.somfytahoma/src/main/java/org/openhab/binding/somfytahoma/internal/handler/SomfyTahomaBaseThingHandler.java
Comment (lolodomo) [reply bundles/org.openhab.binding.somfytahoma/src/main/java/org/openhab/binding/somfytahoma/internal/handler/SomfyTahomaBaseThingHandler.java]: Log removed

**Review by fwolter — APPROVED** [review]
LGTM, thanks!

File: bundles/org.openhab.binding.somfytahoma/src/main/java/org/openhab/binding/somfytahoma/internal/handler/SomfyTahomaBaseThingHandler.java
Comment (fwolter) [reply bundles/org.openhab.binding.somfytahoma/src/main/java/org/openhab/binding/somfytahoma/internal/handler/SomfyTahomaBaseThingHandler.java]: I think it's a documentation issue. I filed a pull-request openhab/openhab-docs#1510.

**Pull request merged**
